### DISCUSSION TITLE
Supporting multilingual text and collaborative translation?

### DISCUSSION THREAD

**Discussion by lapidus** [ueberdosis/tiptap#2530] [General Questions]
Did anyone look into an i18n setup with Tiptap?

Our use case is something along the lines of:
1. An editor creates a Page in a base language (typically English), featuring all sorts of standard and custom blocks and text formatting. 
2. The Page structure is saved to database. Original creators can come back and edit the original language version.
3. Volunteers log into the Page Editor interface and can start **translating the content, keeping the structure of the Page intact**.

From earlier attempts at this – using SlateJS – we stumbled into these complexities:

1. How to store the multilingual text content?

```
{ 
  type: "text",
  textI18N : { en: "Hello!", es: "¡Hola!", ... }
  marks : { ... }
```

2. Given different languages differ in grammar and syntax, how could one add enough flexibility for translators, while retaining as much of the original intent/structure as possible? 

Any ideas appreciated on how to approach this using Tiptap!

ps. My first post here. Nice work on Tiptap! 🌟

**Comment by thefinnomenon** [comment]
The only solution I can think of is when a person wants to translate a page, say to Spanish, a new copy is created in the database and their translations are saved there. If people are translating this original page, I would hope it isn't changing much so having people translate off a duplicate seems like it should work. This would then give you the collaboration and everything out of the box (e.g. all the Spanish translators collaborating in the Spanish copy). If you want to get fancy, you could probably mark blocks with unique IDs, load all the language versions, & use the IDs to show the translations available for the blocks. I'm not exactly sure if this is close to what you are looking for but figured I would throw it out there!

**Comment by lapidus** [comment]
Thank you for the input! 

Saving a new copy is indeed an option, but I am a bit concerned about divergence. In the case that the original page is further edited (structurally), the copies (many languages) would get out of sync?

Ideally the translations would operate more at the block level so that the structure of the page was the same across all languages whereas the i18n content would differ. But it does seem like a very complex case to crack, so I might investigate the "copy page" approach further.

Thanks again!

**Reply by thefinnomenon** [reply]
Yeah the block level would be better but you would still have to deal with the case where the block is edited and I'm sure how you would keep that in sync. Although, if the blocks are only rearranged then it is not an issue. I would say no matter which approach, keeping the content in sync between edits will be difficult.

This is a good example of a structure to handle it on the block level,
https://codesandbox.io/s/r5mve
They use an i18nKey to link the blocks to their translations. Again, I would do something similar but with a random ID for each block which links to the respective translation depending on what editing language they are using. I am not sure how this could easily be integrated into TipTap though.

**Comment by SalahAdDin** [comment]
I'm our case, we only need to translate the editor toolbar, nothing more.
